radiasoft/beamsim/codes/elegant.sh

```diff
@@ -1,3 +1,90 @@
 #!/bin/bash
 codes_dependencies sdds
 codes_download https://depot.radiasoft.org/foss/elegant-28.1.0-1.fedora.21.openmpi.x86_64.rpm
+
+ELEGANT_DOCS=/usr/share/doc/elegant
+ELEGANT_LICENSE=$ELEGANT_DOCS/LICENSE
+TMP_LICENSE=$(mktemp)
+
+cat > $TMP_LICENSE << EOF
```

Comment:
Since you need vars, just enclose in a local name space. (Perhaps `code.sh` should include all imports in a separate function, like `require.js`.) Local vars can be short, because local space so we know that. There's no descriptive value to using long namesl.

```sh
elegant_license() {
    local d=/usr/share/doc/elegant
    if [[ -d $d ]]; then
       # Make re-entrant so users can execute file to update elegant
       return
    fi
    sudo mkdir "$d"
# Quote vars except in variable assignment or [[ ]] or other places known to not need it.
# Use 'EOF', not EOF, so don't get accidental interpolation.
    sudo dd of="$d/LICENSE" <<'EOF'
[...license...]
EOF
    sudo chmod -R root: "$d"
    sudo chmod -R a+rX "$d"
}

elegant_license()
```